plugins/modules/azure_rm_storageaccount.py

```diff
@@ -706,8 +705,7 @@ def update_account(self):
                 except Exception as exc:
                     self.fail("Failed to update account type: {0}".format(str(exc)))
 
-        if self.allow_blob_public_access is not None:
-            if bool(self.allow_blob_public_access) != bool(self.account_dict.get('allow_blob_public_access')):
+        if self.allow_blob_public_access is not None and self.allow_blob_public_access != self.account_dict.get('allow_blob_public_access'):
                 self.results['changed'] = True
```

Comment:
@paultaiton  over indented! Thanks!

Reply:
Got it @Fred-sun . Somehow my linter got disabled. Thanks.